include/seqan/align_parallel/wavefront_task_util.h

```diff
@@ -0,0 +1,554 @@
+
+    alignas(sizeof(TSimdVec)) std::array<TVecVal, LENGTH<TSimdVec>::VALUE> scoreVec;
+    alignas(sizeof(TSimdVec)) std::array<TVecVal, LENGTH<TSimdVec>::VALUE> scoreHorVec;
+    alignas(sizeof(TSimdVec)) std::array<TVecVal, LENGTH<TSimdVec>::VALUE> scoreVerVec;
+    alignas(sizeof(TSimdVec)) std::array<TVecVal, LENGTH<TSimdVec>::VALUE> traceVec;
+
+    auto zipCont = makeZipView(tasks, scoreVec, scoreHorVec, scoreVerVec, traceVec, offset);
+
+    std::for_each(begin(zipCont), end(zipCont),
+                  [&, getBuffer = std::move(getBuffer)](auto tuple)
+                  {
+                      auto& buffer = *getBuffer(*std::get<0>(tuple));
+                      auto val = (length(buffer) > pos) ? buffer[pos] : typename std::decay<decltype(buffer[0])>::type{};
+                      using TDPCellVar = decltype(val.i1);
+                      using TDPCell16 = DPCell_<TVecVal, AffineGaps>;
+
+                      // We might access values out of bounds here.
+                      std::get<1>(tuple) = static_cast<TVecVal>(val.i1._score - std::get<5>(tuple));
+
+                      std::get<2>(tuple) =
+                        (val.i1._horizontalScore <= DPCellDefaultInfinity<TDPCellVar>::VALUE) ?
+                            DPCellDefaultInfinity<TDPCell16>::VALUE :
+                            static_cast<TVecVal>(val.i1._horizontalScore - std::get<5>(tuple));
+                      std::get<3>(tuple) =
+                        (val.i1._verticalScore <= DPCellDefaultInfinity<TDPCellVar>::VALUE) ?
+                        DPCellDefaultInfinity<TDPCell16>::VALUE :
+                        static_cast<TVecVal>(val.i1._verticalScore - std::get<5>(tuple));
+                      std::get<4>(tuple) = val.i2;
+                  });
+
+    target.i1._score = load<TSimdVec>(&scoreVec[0]);
+    target.i1._horizontalScore = load<TSimdVec>(&scoreHorVec[0]);
+    target.i1._verticalScore = load<TSimdVec>(&scoreVerVec[0]);
+    target.i2 = load<TSimdVec>(&traceVec[0]);
+}
+
+template <typename TTasks,
+          typename TDPCell, typename TTrace,
+          typename TPos,
+          typename TFunc,
+          typename TOffset>
+inline void
+storeIntoBuffer(TTasks & tasks,
+                Pair<TDPCell, TTrace> const & source,
+                TPos const pos,
+                TFunc && getBuffer,
+                TOffset const & offset,
+                LinearGaps const & /*unsused*/)
+{
+    using TSimdVec = typename Value<TDPCell>::Type;
+    using TVecVal = typename Value<TSimdVec>::Type;
+
+    alignas(sizeof(TSimdVec)) std::array<TVecVal, LENGTH<TSimdVec>::VALUE> scoreVec;
+    alignas(sizeof(TSimdVec)) std::array<TVecVal, LENGTH<TSimdVec>::VALUE> traceVec;
+
+    storeu(&scoreVec[0], source.i1._score);
+    storeu(&traceVec[0], source.i2);
+
+    auto zipCont = makeZipView(tasks, scoreVec, traceVec, offset);
+
+    std::for_each(begin(zipCont), end(zipCont),
+    [&, getBuffer = std::move(getBuffer)](auto tuple)
```

Comment:
indent lamda body to function parameter begin to be consisten with you other lambdas above